CRM/Contribute/BAO/Contribution.php

```diff
@@ -3539,6 +3539,10 @@ public static function updateFinancialAccounts(&$params, $context = NULL) {
           // & this can be removed
           return;
         }
+        if (empty($params['trxnParams']['currency'])) {
+          // This is an update so original currency if none passed in.
+          $params['trxnParams']['currency'] = isset($params['currency']) ? $params['currency'] : $params['prevContribution']->currency;
```

Comment:
Or you can set as 
```
$params['trxnParams']['currency'] = CRM_Utils_Array::value('currency', $params, $params['prevContribution']->currency);
```

Reply:
@monishdeb I shied away from that because I didn't know whether $params['currency'] might be deliberately set to be empty & what the implications would be on altering it - you think I should be braver?

Reply:
I don't feel any harm in it, as for update we should implicitly use previous currency value. Otherwise it might pick the default currency, set for the system, which will be inappropriate if it doesn't match ```$params['prevContribution']->currency```. What ya say?